### PR TITLE
Threading branch

### PR DESCRIPTION
**Pull request by firzok**

### CODE REVIEW

File: simulation/game_dir/game.py [hunk 1]
```diff
@@ -13,7 +16,13 @@ def __init__(self):
         self.__renderer = Renderer(self.__world)
 
     def run(self):
-        self.renderer.run_simulation()
+
+        # for i in range(1000):
+        event = threading.Event()
+        threading.Thread(target=self.world.update, args=(event,)).start()
```
Comment (internaccounts123) [inline simulation/game_dir/game.py:22]: reverse this please simulation should run in the main thread and renderer should run in the secondary thread.

File: simulation/renderer/adapter.py [hunk 2]
```diff
@@ -70,3 +73,24 @@ def scaling(value, new_max, new_min):
...
             return np.linalg.norm(np.array(ending_pos)-np.array(starting_pos))
+
+    @staticmethod
+    def zoom_in(screen_width, screen_height):
+        Adapter.old_min = [-40, 0]
+        Adapter.old_max = [64, 60]
+        Adapter.new_min = [-400, 0]
+        Adapter.new_max = [1000, 1000]
+        screen_height = np.abs(Adapter.new_max[1] - Adapter.new_min[1])
+        screen_width = np.abs(Adapter.new_max[0] - Adapter.new_min[0])
```
Comment (internaccounts123) [inline simulation/renderer/adapter.py]: comment this out because this won't be necessary.

File: simulation/world/world.py [hunk 3]
```diff
@@ -31,10 +32,22 @@ def cars(self, cars):
...
+    def update(self, event):
+        print(type(event))
+        for i in range(100):
+            event.wait()
+            for car in self.cars:
+                car.y += 1
+                car.front_point = (car.front_point[0], car.front_point[1] + 1)
+                car.back_point = (car.back_point[0], car.back_point[1] + 1)
+            event.clear()
+            print("car updated")
```
Comment (internaccounts123) [inline simulation/world/world.py]: remove print

File: simulation/world/world.py [hunk 4]
```diff
@@ -31,10 +32,22 @@ def cars(self, cars):
     def world_map (self):
         return self.__world_map
 
-    @world_map .setter
+    @world_map.setter
     def world_map (self, world_map ):
         self.__world_map = world_map
 
+    def update(self, event):
+        print(type(event))
```
Comment (internaccounts123) [inline simulation/world/world.py]: remove print

File: simulation/renderer/Renderer.py [hunk 5]
```diff
@@ -131,24 +133,44 @@ def run_simulation(self):
...
             # ----- event handler -----
             for event in pygame.event.get():
                 if event.type == pygame.QUIT:
                     mainloop = False  # window closed by user
                 elif event.type == pygame.KEYDOWN:
                     if event.key == pygame.K_ESCAPE:
                         mainloop = False  # user pressed ESC
+
+                if event.type == pygame.MOUSEBUTTONDOWN:
+                    print(event)
```
Comment (internaccounts123) [inline simulation/renderer/Renderer.py]: remove print

File: simulation/world/world.py [hunk 6]
```diff
@@ -31,10 +32,22 @@ def cars(self, cars):
     def world_map (self):
         return self.__world_map
 
-    @world_map .setter
+    @world_map.setter
     def world_map (self, world_map ):
         self.__world_map = world_map
 
+    def update(self, event):
```
Comment (internaccounts123) [inline simulation/world/world.py:39]: what is the event in this case.

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,17 @@
+.DS_Store
+.idea/
+common/.DS_Store
+common/__pycache__/
```
Comment (internaccounts123) [inline .gitignore]: no need to add .gitignore

**Review by Nabeel07925 — APPROVED** [review]
[OK]

**Pull request merged**